added the canonical tag

jrchudy opened the pull request:
This PR adds the canonical tag to each app `index.html` as mentioned in issue #1798 . This PR also adds a new `chaiseConfig` property called `includeCanonicalTag`. The canonical tag will only show in the html templates if `includeCanonicalTag=true`.

RFSH commented on common/utils.js:
```diff
@@ -1724,6 +1737,18 @@
+            if (chaiseConfig['includeCanonicalTag'] == true) {
+                var canonicalTag = document.createElement("link");
+                canonicalTag.setAttribute("rel", "canonical");
+                var canonicalURL = $window.location.origin + $window.location.pathname + UriUtils.stripSortAndQueryParams($window.location.hash);
```
The window location might be using `?` for the hash portion of the url (remember the changes that we made to the UriUtils to make sure they call the same function that is taking care of this logic).
But why do you need to reconstruct the whole url piece by piece? Why are you not passing the whole `$window.location.href` to the `stripSortAndQueryParams` function that you've wrote?
Then that function should take care of the rest.

jrchudy replied:
I added strip query params after I had done the first part and didn't realize I could just call it on the whole url.

jrchudy commented:
I made some changes based on your comment. I added a function that checks for the location hash and returns the proper value if `#` was replaced with `?`.

I didn't change what was passed to `stripSortAndQueryParams()` because of the `getHash()` function I mentioned I added. I couldn't reuse `chaiseURItoErmrestURI` because it throws an error when there's no catalog, schema, or table defined. I believe, we need this tag for static pages, as well as the deriva webapps (which may not have catalog, schema, or table defined).
